src/models/statement/mod.rs

```diff
@@ -0,0 +1,153 @@
+pub mod statement_naive;
+pub mod statement_v01;
+use serde_json::Value;
+pub use statement_naive::StateNaive;
+pub use statement_v01::StateV01;
+
+use std::convert::{TryFrom, TryInto};
+use std::fmt::{Display, Formatter, Result as FmtResult};
+
+use serde::de::{Deserialize, Deserializer, Error as DeserializeError};
+use serde::ser::{Error as SerializeError, Serialize, Serializer};
+use serde_derive::Serialize;
+use strum::IntoEnumIterator;
+use strum_macros::EnumIter;
+
+use super::link::metadata::MetadataFlatten;
+use super::{Metadata, PredicateWrapper};
+use crate::Error;
+use crate::Result;
+
+#[derive(Debug, Hash, PartialEq, Eq, Clone, Copy, EnumIter)]
+pub enum StatementVer {
+    Naive,
+    V0_1,
+}
+
+impl TryFrom<String> for StatementVer {
+    type Error = crate::Error;
+
+    fn try_from(target: String) -> Result<Self> {
+        match target.as_str() {
+            "link" => Ok(StatementVer::Naive),
+            "https://in-toto.io/Statement/v0.1" => Ok(StatementVer::V0_1),
+            _ => Err(Error::StringConvertFailed(target)),
+        }
+    }
+}
+
+impl TryInto<String> for StatementVer {
+    type Error = crate::Error;
```

Comment:
Why the design here impls trait `TryInto` instead of `Into`? I see there is no error cases.